include/fmt/compile.h

```diff
@@ -463,6 +463,40 @@ template <typename Char, typename T, int N> struct field {
 template <typename Char, typename T, int N>
 struct is_compiled_format<field<Char, T, N>> : std::true_type {};
 
+// A replacement field that refers to argument with name.
+template <typename Char> struct runtime_named_field {
+  using char_type = Char;
+  basic_string_view<Char> name;
+
+  template <typename OutputIt, typename T>
+  constexpr static bool try_format_argument(OutputIt& end, OutputIt out,
+                                            basic_string_view<Char> arg_name,
+                                            const T& arg) {
+    if constexpr (!is_named_arg<typename std::remove_cv<T>::type>::value) {
+      return false;
+    } else {
+      if (arg_name == arg.name) {
```

Comment:
What if there multiple arguments with the same name?

Reply:
Well... I guess the first one will be chosen 🙂. Probably I need to investigate how does runtime API handles that, but right now, it's just implemented in the easiest way possible.

Reply:
It looks like runtime API also chooses the first argument with the specified name - https://godbolt.org/z/8PWEvo.